packages/vaex-ml/vaex/ml/metrics.py

```diff
@@ -1,15 +1,89 @@
+import warnings
+
 import numpy as np
 
 import vaex
 
 
+def _ensure_numpy_array(x):
+    return np.array([x]) if not isinstance(x, np.ndarray) else x
+
 def ensure_string_arguments(*args):
     result = []
     for arg in args:
         result.append(vaex.utils._ensure_string_from_expression(arg))
     return result
 
 
+def _prf_divide(numerator, denominator, metric, modifier, average, warn_for, zero_division="warn"):
+    '''Performs division and handles divide-by-zero.
+
+    On zero-division, sets the corresponding result elements equal to 0 or 1 (according to ``zero_division``).
+    Plus, if ``zero_division != "warn"`` raises a warning.
+
+    The metric, modifier and average arguments are used only for determining an appropriate warning.
+    (this function is taken verbatim from scikit-learn)
+    '''
+    mask = denominator == 0.0
```

Comment:
```suggestion
    zero_mask = denominator == 0.0
```

Reply:
We agree (offline) to leave this function as is, since it is taken verbatim from scikit-learn (docstrings make this clear).